src/main/java/org/commcare/suite/model/Entry.java

```diff
@@ -93,36 +92,8 @@ public Vector<SessionDatum> getSessionDataReqs() {
         return data;
     }
 
-    public Hashtable<String, DataInstance> getInstances() {
-        return getInstances(null);
-    }
-    /**
-     *
-     * @param limitingList a list of instance names to restrict the returning set to; null
-     *                     if no limiting is being used
-     * @return a hashtable representing the data instances that are in scope for this Entry,
-     * potentially limited by @limitingList
-     */
-    public Hashtable<String, DataInstance> getInstances(Set<String> limitingList) {
-        Hashtable<String, DataInstance> copy = new Hashtable<>();
-        for (Enumeration en = instances.keys(); en.hasMoreElements(); ) {
-            String key = (String)en.nextElement();
-
-            //This is silly, all of these are external data instances. TODO: save their
-            //construction details instead.
-            DataInstance cur = instances.get(key);
-            if (limitingList == null || limitingList.contains(cur.getInstanceId())) {
-                // Make sure we either aren't using a limiting list, or the instanceid is in the list
-                if (cur instanceof ExternalDataInstance) {
-                    //Copy the EDI so when it gets populated we don't keep it dependent on this object's lifecycle!!
-                    copy.put(key, new ExternalDataInstance(((ExternalDataInstance)cur).getReference(), cur.getInstanceId()));
-                } else {
-                    copy.put(key, cur);
-                }
-            }
-        }
-
-        return copy;
+    public Hashtable<String, DataInstance> getInstances(Set<String> instancesToInclude) {
+        return InstanceUtils.getLimitedInstances(null, instances);
```

Comment:
should not this be `InstanceUtils.getLimitedInstances(instancesToInclude, instances)` ? Same comment for menu.java

Reply:
fixed here https://github.com/dimagi/commcare-core/pull/1267/commits/338f5f012e7808784e50346ed9662dadd91a710e 
tests also updated in that commit